relayer/metrics/metricsService.go

```diff
@@ -0,0 +1,175 @@
+package metrics
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/lavanet/lava/utils"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+)
+
+type AggregatedMetric struct {
+	TotalLatency int64
+	RelaysCount  uint64
+	SuccessCount int64
+}
+
+type MetricService struct {
+	numberOfRelays      int64
+	aggregatedMetricMap *map[string]map[string]map[string]*AggregatedMetric
+	metricsChannel      chan RelayAnalytics
+	sqsUrl              string
+}
+
+func NewMetricService() *MetricService {
+	metricPortalSqsUrl := os.Getenv("METRICS_PORTAL_SQS_URL")
+	intervalData := os.Getenv("METRICS_INTERVAL_FOR_SENDING_DATA_INM")
+	if metricPortalSqsUrl == "" || intervalData == "" {
+		return nil
+	}
+	intervalForMetrics, _ := strconv.ParseInt(intervalData, 10, 32)
+	utils.LavaFormatDebug("env variables:", &map[string]string{
+		"url":      metricPortalSqsUrl,
+		"interval": intervalData})
+	mChannel := make(chan RelayAnalytics)
+	result := &MetricService{
+		metricsChannel:      mChannel,
+		sqsUrl:              metricPortalSqsUrl,
+		aggregatedMetricMap: &map[string]map[string]map[string]*AggregatedMetric{},
+	}
+	//setup reader
+	go result.ReadFromChannel()
+	//setup sending of the results into the query
+	ticker := time.NewTicker(time.Duration(intervalForMetrics * time.Minute.Nanoseconds()))
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				{
+					data := result.ConvertDataToArray()
+					err := result.SendTo(data)
+					if err != nil {
+						utils.LavaFormatError("error sending metrics data portal app.", err, nil)
+					}
+				}
+			}
+		}
+	}()
+	return result
+}
+
+func (m *MetricService) SendTo(data []RelayAnalyticsDTO) error {
+	if data == nil || len(data) == 0 {
+		return nil
+	}
+	jsonValue, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+	resp, err := http.Post(m.sqsUrl, "application/json", bytes.NewBuffer(jsonValue))
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("couldn't send the data into the server")
+	}
+	return nil
+}
+
+func (m *MetricService) ConvertDataToArray() []RelayAnalyticsDTO {
+	if m.numberOfRelays == 0 || m.aggregatedMetricMap == nil {
+		return nil
+	}
+	toSendData := make([]RelayAnalyticsDTO, m.numberOfRelays)
```

Comment:
if you use append() you don't have to accumulate the numberOfRelays, and have one less dependency/variable to clear